src/corehost/cli/fxr/fx_muxer.cpp

```diff
@@ -659,45 +677,71 @@ bool fx_muxer_t::resolve_sdk_dotnet_path(const pal::string_t& own_dir, const pal
         hive_dir.push_back(global_dir);
     }
 
-    pal::string_t retval;
+    bool cli_version_specified = false;
+    pal::string_t cli_version;
+    pal::string_t sdk_path;
+    pal::string_t global_cli_version;
+
+    if (!global.empty())
+    {
+        global_cli_version = resolve_cli_version(global);
+        if (!global_cli_version.empty())
+        {
+            cli_version_specified = true;
```

Comment:
Is this bool needed or is it invariantly `global_cli_version.empty()`. I'd eliminate the extra state if possible.